tbpore/cli.py

```diff
@@ -29,3 +47,214 @@ def handle_parse_result(self, ctx, opts, args):
                 else:
                     self.prompt = None
         return super(Mutex, self).handle_parse_result(ctx, opts, args)
+
+
+@click.command()
+@click.help_option("--help", "-h")
+@click.version_option(__version__, "--version", "-V")
+@click.option(
+    "-o",
+    "--outdir",
+    help="Directory to place output files",
+    default=".",
+    show_default=True,
+    type=click.Path(file_okay=False, writable=True, path_type=Path),
+)
+@click.option(
+    "-v",
+    "--verbose",
+    help="Turns on debug-level logger. ",
+    is_flag=True,
+    cls=Mutex,
+    not_required_if=["quiet"],
+)
+@click.option(
+    "-q",
+    "--quiet",
+    help="Turns off all logging except errors. ",
+    is_flag=True,
+    cls=Mutex,
+    not_required_if=["verbose"],
+)
+@click.option(
+    "-r", "--recursive", help="Recursively search INPUTS for fastq files", is_flag=True
+)
+@click.option(
+    "--tmp",
+    help=(
+        f"Specify where to write all (tbpore) temporary files. [default: "
+        f"<outdir>/{TMP_NAME}]"
+    ),
+    type=click.Path(file_okay=False, writable=True, path_type=Path),
+)
+@click.option(
+    "-S",
+    "--name",
+    help=(
+        "Name of the sample. By default, will use the first INPUT file with any "
+        "extensions stripped"
+    ),
+)
+@click.option(
+    "-t",
+    "--threads",
+    help="Number of threads to use in multithreaded tools",
+    type=int,
+    default=1
+)
+@click.option(
+    "-m",
+    "--mem_mb",
+    help="Memory to use in tools that accept a memory limit",
+    type=int,
```

Comment:
Might be more ergonomic to use https://github.com/mbhall88/MemoryUnits. It's a "header-only" python library I wrote.

Reply:
Actually, this is probably not worth the effort. Ignore this

Reply:
I will actually remove this argument I think, as it is just `mykrobe` who uses it. We can put a fix 4 or 8 GB value. I am also thinking of having a `config.yaml` file that will configure these magical numbers (and lots of magical parameters in the code)

Reply:
fine by me